2.1.4 Compilation error on Amazon Linux 1  loop initial declarations are only allowed in C99 mode

## Environment

**Operating System**

Amazon Linux 1 (`4.14.225-121.362.amzn1.x86_64`) with GCC 4.8 (`x86_64-amazon-linux gcc version 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC)`)

**TinyTDS Version and Information**

Problem occurs only on tiny_tds v 2.1.4.

```
# tsql -C
Compile-time settings (established with the "configure" script)
                            Version: freetds v1.00.112
             freetds.conf directory: /usr/local/etc
     MS db-lib source compatibility: no
        Sybase binary compatibility: no
                      Thread safety: yes
                      iconv library: yes
                        TDS version: 7.4
                              iODBC: no
                           unixodbc: no
              SSPI "trusted" logins: no
                           Kerberos: no
                            OpenSSL: yes
                             GnuTLS: no
                               MARS: no
```


**FreeTDS Version**

Any, tested on FreeTDS 1.2.21 , 1.1.40, 1.0.86

## Description

```
result.c:107:7: error: ‘for’ loop initial declarations are only allowed in C99 mode
       for (short int j = i; j < userdata->nonblocking_errors_length; j++) {
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 ## (unreleased)
 
+* Fix compilation errors for Amazon Linux 1. Fixes #495.
+
 ## 2.1.4
 
 * Improve handling of network related timeouts

--- a/ext/tiny_tds/result.c
+++ b/ext/tiny_tds/result.c
@@ -99,12 +99,14 @@ static void nogvl_cleanup(DBPROCESS *client) {
   Now that the blocking operation is done, we can finally throw any
   exceptions based on errors from SQL Server.
   */
-  for (short int i = 0; i < userdata->nonblocking_errors_length; i++) {
+  short int i;
+  for (i = 0; i < userdata->nonblocking_errors_length; i++) {
     tinytds_errordata error = userdata->nonblocking_errors[i];
 
     // lookahead to drain any info messages ahead of raising error
     if (!error.is_message) {
-      for (short int j = i; j < userdata->nonblocking_errors_length; j++) {
+      short int j;
+      for (j = i; j < userdata->nonblocking_errors_length; j++) {
         tinytds_errordata msg_error = userdata->nonblocking_errors[j];
         if (msg_error.is_message) {
           rb_tinytds_raise_error(client, msg_error);

